asllib/doc/ASLAbstractSyntaxReference.tex

```diff
@@ -170,4 +219,251 @@ \chapter{ASL Abstract Syntax}
   & & \ASTComment{A name, its corresponding slice, and the type of the bitfield.}\\
 \end{array}
 \]
+
+\[
+\begin{array}{rcl}
+\Field &::=& (\identifier, \ty)\\
+  & & \ASTComment{A field of a record-like structure.}\\
+\typedidentifier &::=& (\identifier, \ty)\\
+  & & \ASTComment{An identifier declared with its type.}\\
+\end{array}
+\]
+
+\[
+\begin{array}{rcl}
+& & \ASTComment{Type of left-hand side of assignments.}\\
+\hline
+\lexpr &::=& \texttt{LE\_Discard}\\
+  & & \ASTComment{\texttt{"-"}}\\
+  &|& \texttt{LE\_Var}(\identifier)\\
+  &|& \texttt{LE\_Slice}(\lexpr, \slice^*)\\
+  &|& \texttt{LE\_SetArray}(\lexpr, \expr)\\
+  &|& \texttt{LE\_SetField}(\lexpr, \identifier)\\
+  &|& \texttt{LE\_SetFields}(\lexpr, \identifier^*)\\
+  &|& \texttt{LE\_Destructuring}(\lexpr^*)\\
+  &|& \texttt{LE\_Concat}(\lexpr^*, \texttt{<int\_lit>}^*?)\\
+  & & \ASTComment{\texttt{LE\_Concat}$(les, \_)$ unpacks the various lexpr.}\\
+  & & \ASTComment{The second argument is a type annotation.}\\
```

Comment:
As far as I see, the second component is never used by the type checker. It is only used by the interpreter. So, from the type system perspective it is redundant.
The thing is, inferring a constant length is not always possible (if an under-constrained integer is used) so this would fail. So, I am confused about why we need this especially that the type checker may fail on it, for seemingly no good reason.
Am I missing something?